src/snowflake/cli/plugins/snowpark/package_utils.py

```diff
@@ -68,165 +66,183 @@ def deduplicate_and_sort_reqs(
+    anaconda: AnacondaChannel | None,
     requirements_file: SecurePath | None,
     packages_dir: SecurePath,
-    perform_anaconda_check: bool = True,
+    perform_anaconda_check_for_dependencies: bool = True,
     package_name: str | None = None,
-    allow_native_libraries: PypiOption = PypiOption.ASK,
+    index_url: str | None = None,
+    allow_shared_libraries: PypiOption = PypiOption.ASK,
+    skip_version_check: bool = False,
 ) -> tuple[bool, SplitRequirements | None]:
     """
-    Install packages from a requirements.txt file or a single package name,
-    into a local folder named '.packages'.
-    If a requirements.txt file is provided, they will be installed using pip.
-    If a package name is provided, it will be installed using pip.
+    Downloads packages from a requirements.txt file or a single package name,
+    into a local directory named '.packages'.
+    If perform_anaconda_check_for_dependencies is set to True, dependencies available in Snowflake Anaconda
+    channel will be omitted, otherwise all packages will be downloaded using pip.
+
     Returns a tuple of:
-    1) a boolean indicating whether the installation was successful
+    1) a boolean indicating whether the download was successful
     2) a SplitRequirements object containing any installed dependencies
-    which are available on the Snowflake Anaconda channel. These will have
-    been deleted from the local packages folder.
+        which are available on the Snowflake Anaconda channel.
+        They will not be downloaded into '.packages' directory.
     """
     if requirements_file and package_name:
         raise ClickException(
             "Could not use package name and requirements file simultaneously"
         )
-
     if requirements_file and not requirements_file.exists():
         raise ClickException(f"File {requirements_file.path} does not exists.")
+    if perform_anaconda_check_for_dependencies and not anaconda:
+        raise ClickException(
+            "Cannot perform anaconda checks if anaconda channel is not specified."
+        )
 
-    with Venv() as v:
+    with Venv() as v, SecurePath.temporary_directory() as downloads_dir:
         if package_name:
             # This is a Windows workaround where use TemporaryDirectory instead of NamedTemporaryFile
-            tmp_requirements = SecurePath(v.directory.name) / "requirements.txt"
-            tmp_requirements.write_text(str(package_name))
-            requirements_file = tmp_requirements
-
-        assert type(requirements_file) is SecurePath
-        pip_install_result = v.pip_install(str(requirements_file.path))
-        dependencies = v.get_package_dependencies(str(requirements_file.path))
+            requirements_file = SecurePath(v.directory.name) / "requirements.txt"
+            requirements_file.write_text(str(package_name))
 
-        if pip_install_result != 0:
-            log.info(pip_failed_msg.format(pip_install_result))
+        pip_wheel_result = v.pip_wheel(
```

Comment:
Why we decided to use `pip wheel`?

Reply:
This was the easiest approach to have consistent artifacts and avoid "install" at the same time - pip wheel builds and packs a wheel from whatever distribution is available.
In future we may want to avoid building - for example use pip download for specific architecture - but I figured that we will think more about it then.